bundles/org.openhab.binding.solarmax/src/main/java/org/openhab/binding/solarmax/internal/connector/SolarMaxData.java

```diff
@@ -0,0 +1,253 @@
+    }
+
+    public DecimalType getStartups() {
+        return getIntegerValueFrom(SolarMaxCommandKey.startups);
+    }
+
+    public DecimalType getAcPhase1Current() {
+        return getDecimalValueFrom(SolarMaxCommandKey.acPhase1Current, 0.01);
+    }
+
+    public DecimalType getAcPhase2Current() {
+        return getDecimalValueFrom(SolarMaxCommandKey.acPhase2Current, 0.01);
+    }
+
+    public DecimalType getAcPhase3Current() {
+        return getDecimalValueFrom(SolarMaxCommandKey.acPhase3Current, 0.01);
+    }
+
+    public DecimalType getEnergyGeneratedToday() {
+        return getIntegerValueFrom(SolarMaxCommandKey.energyGeneratedToday, 100);
+    }
+
+    public DecimalType getEnergyGeneratedTotal() {
+        return getIntegerValueFrom(SolarMaxCommandKey.energyGeneratedTotal, 1000);
+    }
+
+    public DecimalType getOperatingHours() {
+        return getIntegerValueFrom(SolarMaxCommandKey.operatingHours);
+    }
+
+    public DecimalType getEnergyGeneratedYesterday() {
+        return getIntegerValueFrom(SolarMaxCommandKey.energyGeneratedYesterday, 100);
+    }
+
+    public DecimalType getEnergyGeneratedLastMonth() {
+        return getIntegerValueFrom(SolarMaxCommandKey.energyGeneratedLastMonth, 1000);
+    }
+
+    public DecimalType getEnergyGeneratedLastYear() {
+        return getIntegerValueFrom(SolarMaxCommandKey.energyGeneratedLastYear, 1000);
+    }
+
+    public DecimalType getEnergyGeneratedThisMonth() {
+        return getIntegerValueFrom(SolarMaxCommandKey.energyGeneratedThisMonth, 1000);
+    }
+
+    public DecimalType getEnergyGeneratedThisYear() {
+        return getIntegerValueFrom(SolarMaxCommandKey.energyGeneratedThisYear, 1000);
+    }
+
+    public DecimalType getCurrentPowerGenerated() {
+        return getIntegerValueFrom(SolarMaxCommandKey.currentPowerGenerated, 0.5);
+    }
+
+    public DecimalType getAcFrequency() {
+        return getDecimalValueFrom(SolarMaxCommandKey.acFrequency, 0.01);
+    }
+
+    public DecimalType getAcPhase1Voltage() {
+        return getDecimalValueFrom(SolarMaxCommandKey.acPhase1Voltage, 0.1);
+    }
+
+    public DecimalType getAcPhase2Voltage() {
+        return getDecimalValueFrom(SolarMaxCommandKey.acPhase2Voltage, 0.1);
+    }
+
+    public DecimalType getAcPhase3Voltage() {
+        return getDecimalValueFrom(SolarMaxCommandKey.acPhase3Voltage, 0.1);
+    }
+
+    public DecimalType getHeatSinkTemperature() {
+        return getIntegerValueFrom(SolarMaxCommandKey.heatSinkTemperature);
+    }
+
+    @Nullable
+    private DecimalType getDecimalValueFrom(SolarMaxCommandKey solarMaxCommandKey, double multiplyByFactor) {
+        if (this.data.containsKey(solarMaxCommandKey)) {
+            String valueString = this.data.get(solarMaxCommandKey);
+
+            if (valueString != null) {
+                int valueInt = Integer.parseInt(valueString, 16);
+                return new DecimalType((float) valueInt * multiplyByFactor);
```

Comment:
Instead of creating a DecimalType, you need to create a QuantityType to make use of Units of Measure.

Reply:
I think I've done this correctly. A more experienced set of eyes on this code would be most appreciated. At least the values are still being returned correctly.